src/azure-cli-core/setup.py

```diff
@@ -59,7 +59,7 @@
     'requests~=2.22',
     'six~=1.12',
     'pkginfo>=1.5.0.1',
-    'azure-mgmt-core==1.2.1',
```

Comment:
`azure-mgmt-core` was pinned by #15780. @jsntcy, is it necessary that we should pin it?

We does deliver `azure-mgmt-core` at https://pypi.org/project/azure-mgmt-core/1.2.1/. nixpkgs is not one of our supported delivery platform. @fengzhou-msft

Reply:
The Azure SDKs are in the transition of track 1 to track 2, there may be unexpected breaking changes. To play safe, we pinned the version for the transition period and a version bump even with the micro version needs full live tests.

Azure CLI relies on too many Azure SDKs and third-party dependencies, it is strongly recommended to use the all-in-one platform packages or install the wheel package in an isolated python environment.

Reply:
> nixpkgs is not one of our supported delivery platform.

Yes, I maintain azure-cli for nixpkgs / NixOS.

> The Azure SDKs are in the transition of track 1 to track 2, there may be unexpected breaking changes. To play safe, we pinned the version for the transition period and a version bump even with the micro version needs full live tests.

This should only be done if there's breaking changes between patch versions. Azure sdk doesn't.

> it is strongly recommended to use the all-in-one platform packages or install the wheel package in an isolated python environment.

This is what I'm providing for nixpkgs. I already applied the correct patch https://github.com/NixOS/nixpkgs/pull/105267/commits/d03b3c4915e557122658ea0698b89ee0c716194e, I'm just upstreaming the changes so that other azure-cli users can benefit from better dependency management.

> To play safe, we pinned the version for the transition period and a version bump even with the micro version needs full live tests.

What I'm referring to is https://github.com/Azure/azure-sdk-for-python/blob/master/doc/dev/mgmt/mgmt_release.md#version in which python sdk is guaranteeing not to break downstream packages until the next the next major bump

cc @lmazuel

Reply:
> What I'm referring to is https://github.com/Azure/azure-sdk-for-python/blob/master/doc/dev/mgmt/mgmt_release.md#version in which python sdk is guaranteeing not to break downstream packages until the next the next major bump

It's supposed to be, but not always true when developers are not aware of the breaking changes especially with Track 2 adoption. Add @jsntcy who works with this SDK to evaulate if we're confident to follow semver at this stage.